kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/KuraInterfaceStatus.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm.configuration;
```

Comment:
Is it possible to put together this class and the NMDeviceEnable? Maybe a better name is required.

Reply:
I cannot separate the two concepts since one (`NMDeviceEnable`) models whether the device should be disabled/unmanaged/activated, the other (`KuraInterfaceStatus`) models whether the IPv[X] configuration should be set as LAN/WAN/deactivated.

They act at two different abstraction levels:
- `NMDeviceEnable` at `device` a.k.a. `interface` level
- `KuraInterfaceStatus` at IP settings level

I do agree about using a better naming scheme.

Here's my proposal:
- `NMDeviceEnable` -> `KuraDeviceStatus`
- `KuraInterfaceStatus` -> `KuraIpStatus`

Reply:
For me it's ok. Probably I prefer:

- `NMDeviceEnable` -> `KuraInterfaceStatus`
- `KuraInterfaceStatus` -> `KuraIpStatus`

`device` looks like the whole system...

Reply:
I used `device` because NM calls them that way, but, since we're modeling Kura concepts, it makes sense to use the Kura naming convention (i.e. `interface`).